planning/obstacle_avoidance_planner/src/utils/utils.cpp

```diff
@@ -708,6 +712,37 @@ bool isOutsideDrivableAreaFromRectangleFootprint(
     tier4_autoware_utils::calcOffsetPose(traj_point.pose, -base_to_rear, -base_to_left, 0.0)
       .position;
 
+  if (is_considering_footprint_edges) {
+    LinearRing2d footprint_polygon;
+    LineString2d left_bound_line;
+    LineString2d right_bound_line;
+    LineString2d back_bound_line;
+
+    footprint_polygon.push_back({top_left_pos.x, top_left_pos.y});
+    footprint_polygon.push_back({top_right_pos.x, top_right_pos.y});
+    footprint_polygon.push_back({bottom_right_pos.x, bottom_right_pos.y});
+    footprint_polygon.push_back({bottom_left_pos.x, bottom_left_pos.y});
+
+    bg::correct(footprint_polygon);
+
+    for (const auto & p : left_bound) {
+      left_bound_line.push_back({p.x, p.y});
+    }
+
+    for (const auto & p : right_bound) {
+      right_bound_line.push_back({p.x, p.y});
+    }
+
+    back_bound_line = {left_bound_line.back(), right_bound_line.back()};
+
+    if (
+      bg::intersects(footprint_polygon, left_bound_line) ||
+      bg::intersects(footprint_polygon, right_bound_line) ||
+      bg::intersects(footprint_polygon, back_bound_line)) {
+      return true;
+    }
+  }
```

Comment:
@takayuki5168 If there is intersection that means footprint is on the drivable area bounds. So vehicle is the outside drivable area. And because of that function should return true.

Reply:
I meant this one. It seems this line was added after my comment. Resolved.
https://github.com/autowarefoundation/autoware.universe/blob/398d08903e6550e23d021579210d83ca8da0c0b0/planning/obstacle_avoidance_planner/src/utils/utils.cpp#L744